rcl/test/rcl/test_subscription.cpp

```diff
@@ -528,3 +528,150 @@ TEST_F(CLASSNAME(TestSubscriptionFixture, RMW_IMPLEMENTATION), test_get_options)
 
   ASSERT_EQ(NULL, rcl_subscription_get_options(nullptr));
 }
+
+/* Using bad arguments subscription methods
+ */
+TEST_F(CLASSNAME(TestSubscriptionFixture, RMW_IMPLEMENTATION), test_subscription_bad_argument) {
```

Comment:
This test is pretty large, can you split it up into smaller tests?

Reply:
Done!